bokeh/models/mappers.py

```diff
@@ -63,3 +65,229 @@ class LinearColorMapper(ColorMapper):
+
+    def __init__(self, palette=None, segments=None, alpha=1, interpolationMethod='linear', **kwargs):
+        # Elegantly handle the possability that some parameters can be passed from the
+        # kwargs structure (because of subclassing)
+        if palette is None and 'palette' in kwargs.keys():
+            palette = kwargs['palette']
+
+        if segments is None and 'segments' in kwargs.keys():
+            segments = kwargs['segments']
+
+        if alpha is None and 'alpha' in kwargs.keys():
+            alpha = kwargs['alpha']
+
+        if interpolationMethod is None and 'interpolationMethod' in kwargs.keys():
+            interpolationMethod = kwargs['interpolationMethod']
+
+
+        # Define a dict to hold any alpha values that we might extract
+        # from the different colors in the passed palette.  These can
+        # come from a 4-tuple or a
+        alpha_hold = {}
+
+        if palette is not None:
+            tmp_palette = expandPalette(palette)
+            palette = tmp_palette['palette']
+            alpha_hold = tmp_palette['alpha_hold']
+            kwargs['palette'] = palette
+
+        if alpha is not None and palette is not None:
+            if not isinstance(alpha, list):
+                # Convert the scalar object to a vector so that the Seq property
+                # type can be used.
+                alpha = [alpha] * len(palette)
+
+            # Overwrite the alpha values with those specified by the colors in the palette.
+            # This may not be the desired functionality.  Who should win, the information in
+            # the palette, or the direct alpha information specified in the alpha parameter?
+            for k,v in list(alpha_hold.items()):
+                alpha[k] = v
+
+            # Check to ensure that all of the alpha values are ok.
+            if any([x > 1 for x in alpha]):
+                raise ValueError('alpha can not have any values greater than 1.0')
+
+            if any([x < 0 for x in alpha]):
+                raise ValueError('alpha cannot have any values less than 0.0')
+
+            kwargs['alpha'] = alpha
+
+        if segments is not None: kwargs['segments'] = segments
+
+        kwargs['interpolationMethod'] = interpolationMethod
+
+        super(SegmentedColorMapper, self).__init__(**kwargs)
+
+
+class LinearColorMapperReplacement(SegmentedColorMapper):
+
+    low = Float(help="""
+    The minimum value of the range to map into the palette. Values below
+    this are clamped to ``low``.
+    """)
+
+    high = Float(help="""
+    The maximum value of the range to map into the palette. Values above
+    this are clamped to ``high``.
+    """)
+
+    # TODO: (jc) what is the color code for transparent?
+    # TODO: (bev) better docstring
+    reserve_color = Color("#ffffff", help="""
+    Used by Abstract Rendering.
+    """)
+
+    # TODO: (bev) better docstring
+    reserve_val = Float(default=None, help="""
+    Used by Abstract Rendering.
+    """)
```

Comment:
Wellllll.......  Lets not be so quick here.  I think that they serve a nice purpose.  One one hand we have a colormap which can assign colors to values in a specific way.  On the other hand, it would be nice if certain values could be given a one off color, to possibly highlight some special state.  Quick thoughts on use case for this would be highlighting a NaN, setting all values of 10 to a special color.  

Now, I know that this can be effectively handled by setting the appropriate segments and then the colors for those segments, but that is quite a bit of work to get a specific effect.  Lets think on this one a bit before we remove the functionality.  That is just my suggestion.